src/relayserver/RelayServer.ts

```diff
@@ -99,12 +108,26 @@ export class RelayServer extends EventEmitter {
     return this.customReplenish
   }
 
-  async pingHandler (verifier?: string): Promise<PingResponse> {
+  getWorkerIndex (address: PrefixedHexString): number {
+    const workerIndex = this.workerAddress.indexOf(address)
+    if (workerIndex > -1) {
+      return workerIndex
+    } else {
+      throw new Error(
+          `Wrong worker address: ${address}\n`)
+    }
+  }
+
+  async pingHandler (verifier?: string, maxTime?: string): Promise<PingResponse> {
+    const relayWorkerAddress = this.envelopingArbiter.getQueueWorker(this.workerAddress, maxTime)
+    const minGasPrice = await this.envelopingArbiter.getQueueGasPrice(maxTime)
+    const maxDelay = this.envelopingArbiter.checkMaxDelayForResponse(maxTime)
     return {
-      relayWorkerAddress: this.workerAddress,
+      relayWorkerAddress,
       relayManagerAddress: this.managerAddress,
       relayHubAddress: this.relayHubContract?.address ?? '',
-      minGasPrice: this.getMinGasPrice().toString(),
+      minGasPrice,
+      maxDelay,
       chainId: this.chainId.toString(),
```

Comment:
Actually when you have a property that matches with a variable is better if you only put the variable there, since doing things like this `minGasPrice: minGasPrice` is redundant.

Reply:
fair enough